File: FakturusTrack.iOS/FakturusTrack/Models/OfflineSession.swift
import Foundation

struct OfflineSession: Codable {
    let userId: String
    let lastSuccessfulAuth: Date

    var isValid: Bool {
        let maxOfflineDays = 14.0
        let daysSinceAuth = Date().timeIntervalSince(lastSuccessfulAuth) / 86400
        return daysSinceAuth <= maxOfflineDays
    }

    var daysUntilExpiry: Int {
        let daysSinceAuth = Date().timeIntervalSince(lastSuccessfulAuth) / 86400
        return max(0, Int(14.0 - daysSinceAuth))
    }
}

File: FakturusTrack.Android/app/src/main/java/com/fakturus/track/models/OfflineSession.kt
package com.fakturus.track.models

import kotlinx.serialization.Serializable

@Serializable
data class OfflineSession(
    val userId: String,
    val lastSuccessfulAuthEpochMillis: Long
) {
    val isValid: Boolean
        get() {
            val maxOfflineDays = 14
            val daysSinceAuth = (System.currentTimeMillis() - lastSuccessfulAuthEpochMillis) / 86_400_000
            return daysSinceAuth <= maxOfflineDays
        }

    val daysUntilExpiry: Int
        get() {
            val daysSinceAuth = (System.currentTimeMillis() - lastSuccessfulAuthEpochMillis) / 86_400_000
            return maxOf(0, (14 - daysSinceAuth).toInt())
        }
}
